test/uk/gov/hmrc/agentservicesaccount/views/pages/AMLS/UpdateMoneyLaunderingSupervisionDetailsSpec.scala

```diff
@@ -0,0 +1,141 @@
+      }
+      "display correct heading title" in {
+        val errorLink: Element = doc.select("legend").get(0)
+        errorLink.text() mustBe "What are your money laundering supervision registration details?"
+      }
+      "display correct date title" in {
+        val errorLink: Element = doc.select(".govuk-hint").get(1)
+        errorLink.text() mustBe "For example, 31 3 2024"
+      }
+      "display correct date hint" in {
+        val errorLink: Element = doc.select(".govuk-hint").get(1)
+        errorLink.text() mustBe "For example, 31 3 2024"
+      }
+      "display correct error date " in {
+        val errorLink: Element = doc.select(".govuk-date-input").first()
+        errorLink.text() mustBe "Day Month Year"
+      }
+      "have the correct service name link" in {
+        doc.select(".hmrc-header__service-name").first.text() mustBe "Agent services account"
+        doc.select(".hmrc-header__service-name").first.attr("href") mustBe "/agent-services-account"
+      }
+      "have the correct sign out link" in {
+        doc.select(".hmrc-sign-out-nav__link").first.text() mustBe "Sign out"
+        doc.select(".hmrc-sign-out-nav__link").first.attr("href") mustBe "/agent-services-account/sign-out"
+      }
+      "have the correct back link" in {
+        doc.select(".govuk-back-link").first.text() mustBe "Back"
+        doc.select(".govuk-back-link").first.attr("href") mustBe "#"
+      }
+    }
+    "form is submitted with errors should" should {
+      val doc: Document = Jsoup.parse(view.apply(formWithErrors, Map[String, String](elems = "AA" -> "AgentService"))(messages, FakeRequest(), appConfig).body)
+
+      "display correct error summary link" in {
+        val errorLink: Element = doc.select(".govuk-error-summary__list a").first()
+        errorLink.text() mustBe "Enter your money laundering supervision registration number"
+        errorLink.attr("href") mustBe "#number"
+      }
+      "display correct amls codes within the select component" in {
+        val component: Element = doc.select("#body").first()
+        component.select("option").get(0).text() mustBe "AgentService"
+        component.select("option").get(0).attr("value") mustBe "AA"
+      }
+      "display error message on form for amls body" in {
+        val errorLink: Element = doc.select(".govuk-error-message").get(0)
+        errorLink.text() mustBe "Error: Enter your money laundering supervisory body"
+      }
+      "display error message on form for date" in {
+        val errorLink: Element = doc.select(".govuk-error-message").get(2)
+        errorLink.text() mustBe "Error: Enter a real date"
+      }
+      "display error if registration number is invalid or empty" in {
+        val errorLink: Element = doc.select(".govuk-error-message").get(1)
+        errorLink.text() mustBe "Error: Enter your money laundering supervision registration number"
+
+        errorLink.attr("href") mustBe ""
```

Comment:
You don't need this line here, it's not adding any value - the val in this test is only selecting the 2nd error message. It's not a link so it might be helpful to rename the val?

That goes for the rest of your tests too, when assigning to a val try and describe what element you are selecting eg. "errorMessage"

Technically you don't need to assign to a val - you could rewrite it as 
```
doc.select(".govuk-error-message").get(1).text() mustBe "Error: Enter your money laundering supervision registration number"
```
but the use of a val is meant to increase readability so 👀